### PR TITLE
Refactor string splitting

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -219,12 +219,10 @@ def parse_s3_url(s3url: str) -> tuple[str, str]:
         elif format[0] == "https:":
             temp_split = format[1].split(".")
             if temp_split[0] == "s3":
-                split_url = format[1].split("/")
-                bucket_name = split_url[1]
-                key = "/".join(split_url[2:])
+                _, bucket_name, key = format[1].split("/", 2)
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/s3.py:222]: Wouldn’t this be just a subset of `temp_split`? I have a feeling this entire block can be rewritten a bit to split less

File: dev/breeze/src/airflow_breeze/params/doc_build_params.py [hunk 2]
```diff
@@ -20,7 +20,15 @@
...
-from airflow_breeze.utils.general_utils import get_provider_name_from_short_hand
+
+providers_prefix = "apache-airflow-providers-"
+
+
+def get_provider_name_from_short_hand(short_form_providers: tuple[str]):
+    providers = []
+    for short_form_provider in short_form_providers:
+        providers.append(providers_prefix + short_form_provider.replace(".", "-"))
+    return tuple(providers)
```
Comment (uranusjr) [inline dev/breeze/src/airflow_breeze/params/doc_build_params.py]:
```suggestion
    return tuple(
        providers_prefix + short_form_provider.replace(".", "-")
        for short_form_provider in short_form_providers
    )
```

File: scripts/in_container/verify_providers.py [hunk 3]
```diff
@@ -360,7 +360,7 @@ def convert_class_name_to_url(base_url: str, class_name) -> str:
     :param class_name: name of the class
     :return: URL to the class
     """
-    return base_url + os.path.sep.join(class_name.split(".")[:-1]) + ".py"
+    return base_url + class_name.rpartition(".")[0].replace(".", os.path.sep) + ".py"
```
Comment (uranusjr) [inline scripts/in_container/verify_providers.py]:
```suggestion
    return base_url + class_name.rpartition(".")[0].replace(".", "/") + ".py"
```

Not sure if this fits in this PR, but the original code is wrong in using `os.path.sep`. This formats a _URL_ so `/` should always be used instead.

File: tests/providers/amazon/aws/hooks/test_s3.py [hunk 4]
```diff
@@ -1052,10 +1053,7 @@ def test_generate_presigned_url(self, s3_bucket):
         presigned_url = hook.generate_presigned_url(
             client_method="get_object", params={"Bucket": s3_bucket, "Key": "my_key"}
         )
-
-        url = presigned_url.split("?")[1]
-        params = {x[0]: x[1] for x in [x.split("=") for x in url[0:].split("&")]}
-
+        params = parse_qs(presigned_url.split("?")[1])
```
Comment (uranusjr) [inline tests/providers/amazon/aws/hooks/test_s3.py]:
```suggestion
        params = parse_qs(presigned_url.split("?", 1)[1])
```

**Review by uranusjr — APPROVED** [review]

File: tests/test_utils/get_all_tests.py [hunk 5]
```diff
@@ -56,7 +56,7 @@ def print_all_cases(xunit_test_file_path):
     modules = set()
 
     for test_case in test_cases:
-        the_module = ".".join(test_case.get("classname").split(".")[:-1])
+        the_module = test_case.get("classname").rpartition(".")[0]
```
Comment (uranusjr) [inline tests/test_utils/get_all_tests.py]: Hmm this should be

```suggestion
        the_module = test_case["classname"].rpartition(".")[0]
```

But it’s a problem inherited from old code.

**Review by uranusjr — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/s3.py (hunk 1 above)
Comment (eumiro) [reply airflow/providers/amazon/aws/hooks/s3.py:222]: Had to revert this. In one test, there was a dot in the filename.

**Review by uranusjr — APPROVED** [review]

**Pull request merged**